scripts/ci_validate_modification.ts

```diff
@@ -170,20 +170,6 @@ function validateBenchmarkResultMod<T>(diff: Diff<T>, expectedBenchName: string,
         if (suite.tool !== added.tool) {
             throw new Error(`Tool is different between ${JSON.stringify(suite)} and ${JSON.stringify(added)}`);
         }
-
-        for (const addedBench of added.benches) {
-            for (const prevBench of suite.benches) {
-                if (prevBench.name === addedBench.name) {
-                    if (prevBench.unit !== addedBench.unit) {
```

Comment:
I'm thinking an alternative and starting to not like this approach. This seems breaking the general language agnostic data structure of GAB(github action benchmark).

There is a potential "key" concept in GAB that uniquely maps to one metric (which is one diagram in the github page). The "unit" in the data structure should just be pure unit instead of a part of "key". Currently "name" is assumed to be the key. However, in Go benchmarking, "name" + "unit" is actually the key of a metric, so currently logic is not compatible with Go.

One alternative is just mapping "<name>:<unit>" in Go result to "name" in GAB. But it will not directly expose the "name" of go benchmarking in the output data, which is not ideal for the users who want to diaplay the real name in their website.

The other alternative is adding another field "id" to data structure. We can make it "<name>:<unit>" or a hash of it. It requires more change but I feel it is the clean and right way, and lean to it now.

What do you think? @ktrz

Reply:
Yeah this is my first alternative. It is ok but I think the only drawback is if the users want to display the pure name ("BenchmarkFib22") from `go test -bench` in some places (or pass it to other programs), they don't have a straightforward field to use. They may have to parse the name by themselves to extract the pure name. We can add another field for the pure name but "name" seems the best naming for it.

So 3 options here.

**option 1**
Add a new field "id" for the unique key of each metric. "name" will become duplicated possible.


**option 2**
Keep "name" as the unique key of each metric. For Go, append unit to "name".

Cons: if the users want to display the pure name ("BenchmarkFib22") from `go test -bench` in some places (or pass it to other programs), they don't have a straightforward field to use

**option 3**
Keep "name" as the unique key of each metric. For Go, append unit to "name". And add another field to store the pure name of Go.

Cons: It is hard to properly name it and it can be weird to have a field specifically for Go.

Reply:
I would go with option 2 for now. We can expand to option 3 later if that gets requested or maybe we'll find another solution

Reply:
Sounds good to me. Will publish a new revision with option 2.